src/dataflow/src/decode/mod.rs

```diff
@@ -714,3 +754,50 @@ where
     });
     (results, None)
 }
+
+fn to_kafka_metadata(
+    metadata_items: IncludeRequests,
```

Comment:
I'm not sure that the 7 element array limit is ever going to actually be a meaningful issue, my main feeling was in fact that not needing to do that pointer deref for this would be valuable, and also not needing to store take up the extra 24 bytes in the core decode loop for something that is currently guaranteed to always be at most 4 elements long.

It's obviously kind of a stupid micro-optimization and I can understand an argument that it's unnecessarily complex, but the limit doesn't seem like the issue. If you'd rather I get rid of it I can, but it felt ... bad ... to have all that indirection for something that ideally would just be a field in the decoder struct.

Reply:
I've pushed up a commit that does remove `IncludeRequests`, but I think that `IncludeRequests` adds value and I'd rather merge without that change than with it. 

That said, I'd rather merge this than debate it, so I'm fine either way.